packages/google-pay-integration/src/google-pay-paypal-commerce/google-pay-paypal-commerce-script-loader.ts

```diff
@@ -0,0 +1,83 @@
+import { ScriptLoader } from '@bigcommerce/script-loader';
+
+import {
+    MissingDataError,
+    MissingDataErrorType,
+    PaymentMethodClientUnavailableError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { GooglePayConfig, PayPalCommerceHostWindow, PayPalGoogleSdk } from './types';
+
+export default class GooglePayPayPalCommerceScriptLoader {
+    private window: PayPalCommerceHostWindow;
+    private googlepayConfig?: GooglePayConfig;
+    private paypalGoogleSdk?: PayPalGoogleSdk;
+
+    constructor(private scriptLoader: ScriptLoader) {
+        this.window = window;
+    }
+
+    async loadPayPalGooglePaySDK(
+        clientId: string,
+        merchantId: string,
+        clientToken: string,
+    ): Promise<PayPalGoogleSdk> {
+        const scriptSrc = `https://www.paypal.com/sdk/js?components=googlepay&client-id=${clientId}&merchant-id=${merchantId}`;
+
+        await this.scriptLoader.loadScript(scriptSrc, {
+            async: true,
+            attributes: {
+                'data-client-token': clientToken,
+                'data-partner-attribution-id': 'GOOGLEPAY',
+            },
+        });
+
+        if (!this.window.paypal) {
+            throw new PaymentMethodClientUnavailableError();
+        }
+
+        return this.window.paypal;
+    }
+
+    async getPayPalGooglePaySdkOrThrow(
+        clientId?: string,
+        merchantId?: string,
+        clientToken?: string,
+    ) {
+        if (!this.paypalGoogleSdk) {
```

Comment:
To avoid loading new `PayPalGooglePaySdk` on each button load, it might be better to place the PayPalGooglePaySdk to the window object.

Reply:
Thanks for mentioning it! The `PayPalGooglePaySdk` script saves in `window.paypal` object. And `loadPayPalGooglePaySDK` calls only once. So I don't think there will be an issue with multiple `PayPalGooglePaySdk` loading.

<img width="1083" alt="Screenshot 2023-10-16 at 12 19 13" src="https://github.com/bigcommerce/checkout-sdk-js/assets/130665807/26bc512e-6e2d-4e87-92f1-eb5e8f0f1ed6">

Reply:
Yes, but each time when loadPayPalGooglePaySDK called, the new script is added to the page.

Reply:
You can cover first 9 lines of the method with `if (!this.window.paypal) { // load script here }` condition

Reply:
> You can cover first 9 lines of the method with `if (!this.window.paypal) { // load script here }` condition

I got your point, updated the PR. 
I check for `!this.window.paypal?.Googlepay` because we can have a common `paypal` script loaded without `Googlepay` field.